### PR TITLE
Set things on many samples at once

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR adds the possibility to set things on many samples at once

**How to prepare for test**:
- [x] ssh to hasta depending on type of change
- [x] install on stage:
`bash servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

**How to test**:
- [x] do cg --log-level=DEBUG set sample ACC6645A7 -kv data_analysis Balsamic -y --skip-lims

**Expected test outcome**:
- [x] check that no errors reported
- [x] Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by @patrikgrenfeldt 
- [x] tests executed by @patrikgrenfeldt @annaengstrom 
- [x] "Merge and deploy" approved by @patrikgrenfeldt 
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/cli/set.py [hunk 1]
```diff
@@ -54,191 +54,281 @@ def family(context, action, priority, panels, family_id):
...
-    context.obj["status"].commit()
-    click.echo(echo_msg)
+            if not (yes or click.confirm("Continue?")):
+                context.abort()
 
+            context.obj["lims"].update_sample(lims_id=sample_id, **{key: value})
+            click.echo(click.style(f"Set LIMS/{key} to {value}", fg="blue"))
+
+
+# @set_cmd.command()
```
Comment (henrikstranneheim) [inline cg/cli/set.py:142]: Remove comments

File: cg/store/api/findbusinessdata.py [hunk 2]
```diff
@@ -228,6 +228,17 @@ def samples(self, *, customer: models.Customer = None, enquiry: str = None) -> L
         )) if enquiry else records
         return records.order_by(models.Sample.created_at.desc())
 
+    def samples_by_ids(self, **identifiers) -> List[models.Sample]:
+        records = self.Sample.query
+
+        print(type(identifiers), identifiers)
```
Comment (henrikstranneheim) [inline cg/store/api/findbusinessdata.py:234]: Should it print here?

**Comment by henrikstranneheim** [comment]
Is there an AM doc that should be uppdated as well?

File: cg/cli/set.py [hunk 3]
```diff
@@ -59,7 +59,7 @@ def family(context, action, priority, panels, family_id):
 @click.option("--skip-lims", is_flag=True, help="Skip setting value in LIMS")
 @click.option("-y", "--yes", is_flag=True, help="Answer yes on all confirmations")
 @click.pass_context
-def samples(context, identifiers, kwargs, kip_lims, yes):
+def samples(context, identifiers, kwargs, skip_lims, yes):
```
Comment (henrikstranneheim) [inline cg/cli/set.py:62]:
```suggestion
def samples(context, identifiers, kwargs, dry_run, yes):
```
This is dry_run now is it not?

File: cg/cli/set.py [hunk 4]
```diff
@@ -54,191 +54,281 @@ def family(context, action, priority, panels, family_id):
...
-    help="sets number of downsampled \
-              total reads. Enter 0 to reset.",
-)
-@click.option("-a", "--application-tag", "apptag", help="sets application tag.")
-@click.option("-k", "--capture-kit", help="sets capture kit.")
-@click.option("--data-analysis", help="sets data-analysis.")
-@click.option("-n", "--name", help="sets name.")
+@click.option('-id', '--identifier', 'identifiers', nargs=2, type=click.Tuple([str, str]), multiple=True)
+@click.option('-kv', '--key-value', 'kwargs', nargs=2, type=click.Tuple([str, str]), multiple=True)
 @click.option("--skip-lims", is_flag=True, help="Skip setting value in LIMS")
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/cli/set.py:59]: There is no dry-run option so far. The --skip-lims flag will prevent cg from trying to set a value in lims which comes in handy when you want to set things on samples that do not exist in LIMS

File: cg/cli/set.py [hunk 5]
```diff
@@ -54,191 +54,281 @@ def family(context, action, priority, panels, family_id):
...
+@click.option("-y", "--yes", is_flag=True, help="Answer yes on all confirmations")
+@click.pass_context
+def samples(context, identifiers, kwargs, kip_lims, yes):
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/cli/set.py]: fixed, good catch

File: cg/store/api/findbusinessdata.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply cg/store/api/findbusinessdata.py:234]: no it was a debug thing, removed now

File: cg/cli/set.py (hunk 1 above)
Comment (patrikgrenfeldt) [reply cg/cli/set.py]: removed

Comment (patrikgrenfeldt) [reply cg/cli/set.py]: removed

File: cg/cli/set.py (hunk 3 above)
Comment (patrikgrenfeldt) [reply cg/cli/set.py:62]: no it is about changing also in lims or not

**Comment by patrikgrenfeldt** [comment]
Test result: 
image

👍

**Comment by patrikgrenfeldt** [comment]
deployed on hasta:
image

**Comment by annaengstrom** [comment]
The documentation regarding this is not complete. Where could it be? cg README.md or AM Procedure for example?

**Pull request merged**